plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -1432,13 +1434,21 @@ public boolean stop() {
         return true;
     }
 
+    /**
+     * This finds a command wrapper to handle the command and executes it.
+     * If no wrapper is found an {@see UnsupportedAnswer} is sent back.
+     * Any other exceptions are to be caught and wrapped in an generic {@see Answer}, marked as failed.
+     *
+     * @param cmd the instance of a {@see Command} to execute.
+     * @return the for the {@see Command} appropriate {@see Answer} or {@see UnsupportedAnswer}
+     */
     @Override
     public Answer executeRequest(final Command cmd) {
```

Comment:
What about a unit test case? Then, you can have a test case that detects if we catch only `RequestWrapper.CommandNotSupported` exceptions.

Reply:
that would require throwing everything else. I will add a subset to catch the most obvious cases

Reply:
Ah, yes. If you want to check all possible cases. However, you can test with the most obvious ones such as `Nullpointer, Exception, CloudRuntimeException, and RuntimeException`

Reply:
I am not really convinced that any of these make a lot of difference to one and another. How about the known and unknown case I just added?

Reply:
It helps. I only think that you could go and test the method `retryWhenAllFails` instead of using `executeRequest`. If we want unit tests, we should write them for the units.

Reply:
we could also unittest RequestWrapper, yes.
executeRequest is the unit here.
higher prios came up, i might revisit

Reply:
No problems ;)
Thanks anyways for writing at least some tests.